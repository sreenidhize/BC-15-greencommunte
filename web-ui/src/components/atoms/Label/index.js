import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  small: {
    fontSize: 20,
    fontFamily: theme.typography.fontFamily,
  },
  large: {
    fontSize: 26,
    fontFamily: theme.typography.fontFamily,
  },
}));

const Label = (props) => {
  const classes = useStyles();
  const { size, text } = props;
  return (
    <div>
      {size === "large" ? (
        <Typography
          variant="h2"
          className={classes.large}
          data-testid="large-text"
        >
          {text}
        </Typography>
      ) : (
        <Typography
          variant="body2"
          className={classes.small}
          data-testid="small-text"
        >
          {text}
        </Typography>
      )}
    </div>
  );
};

export default Label;
