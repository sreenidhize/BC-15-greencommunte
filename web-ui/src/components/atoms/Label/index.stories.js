import React from "react";
import Label from ".";

export default {
  title: "atoms/LabelPlaceHolder",
  component: Label,
};

const Template = (args) => <Label {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  size: "small",
  text: "Real-Time air quality index (AQI)",
};
