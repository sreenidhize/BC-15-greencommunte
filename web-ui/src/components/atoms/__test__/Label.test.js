import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Label from "../Label";

describe("Label placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
        <Label size="large" text="AQI" />
    );
    expect(container).toMatchSnapshot();
  });

  test("test large text render", () => {
    const { getByTestId } = render(<Label size="large" text="AQI"/>);
    expect(getByTestId("large-text")).toBeInTheDocument();
  });

  test("test small text render", () => {
    const { getByTestId } = render(<Label size="small" text="AQI"/>);
    expect(getByTestId("small-text")).toBeInTheDocument();
  });
});
