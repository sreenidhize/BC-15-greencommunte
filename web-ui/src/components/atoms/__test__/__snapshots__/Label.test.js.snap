// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label placeholder test Matches the snapshot 1`] = `
<div>
  <div>
    <h2
      class="MuiTypography-root makeStyles-large-2 MuiTypography-h2"
      data-testid="large-text"
    >
      AQI
    </h2>
  </div>
</div>
`;
